createrepo_mod/createrepo_mod.py

```diff
@@ -43,10 +44,47 @@ def run_modifyrepo(path, compress_type=None):
     return proc.returncode
 
 
+def find_module_yamls(path):
+    """
+    Recursivelly find modulemd YAML files and return a list of their relative
+    paths.
+    """
+    matches = []
+    for root, dirnames, filenames in os.walk(path):
+        for filename in filenames:
+            if not filename.endswith((".yaml", ".yaml.gz")):
+                continue
+            filepath = os.path.join(root, filename)
+            if not is_yaml_valid_modulemd(filepath):
+                continue
+            matches.append(filepath)
+    return matches
+
+
+def is_yaml_valid_modulemd(path):
+    cmd = ["modulemd-validator", "-q", path]
+    proc = subprocess.run(cmd)
+    return proc.returncode == 0
```

Comment:
While you *can* do things this way, it is probably easier to skip the subprocess and just call:
```
idx = Modulemd.ModuleIndex.new()
(ret, failures) = idx.update_from_file(path, strict=True)
```
If `ret` is `False`, you can iterate through `failures` to print the reasons or if you only care about success or failure, just return `ret`.

Aside from handling CLI arguments and reporting errors, that's pretty much all that modulemd-validator is doing.

Reply:
Thank you for the suggestion.
While it may be a bit faster to do this without `subprocess` I would rather call the script anyway. I've updated the code since this comment - it turned out that the `modulemd-validator` cannot parse `.yaml.gz` files, so I am temporarily skipping the validation for them ... and that feature should probably be implemented. By calling the script, we will avoid implementing it on multiple places.

Reply:
> Thank you for the suggestion.
> While it may be a bit faster to do this without `subprocess` I would rather call the script anyway. I've updated the code since this comment - it turned out that the `modulemd-validator` cannot parse `.yaml.gz` files, so I am temporarily skipping the validation for them ... and that feature should probably be implemented. By calling the script, we will avoid implementing it on multiple places.

Hmm, that's odd... it shouldn't have any problems with gzipped files. Please file a bug. I will fix that immediately.

Reply:
I see the problem. When I added the compression support to `Modulemd.update_from_file()`, I neglected to add it to the validator as well. Oops.

I'd recommend going the route of `update_from_file()` I suggested above for now; it will handle the gzipped files fine. I'll get a fix out for the validator soon.

Reply:
I created an issue here https://github.com/fedora-modularity/libmodulemd/issues/479 and I am going to use the suggested code.

Reply:
Updated, PTAL.